src/mesh/mesh.cpp

```diff
@@ -835,6 +827,13 @@ void Mesh::OutputMeshStructure(int ndim) {
     }
   }
 
+  delete[] nb_per_plevel;
+  delete[] cost_per_plevel;
+
+  if (!dump_mesh_structure) {
+    return;
+  }
```

Comment:
Is there a reason we can't just make these unique_ptrs?

Reply:
done